modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/security/permission/resource/CommerceOrderPortletResourcePermissionLogic.java

```diff
@@ -5,22 +5,63 @@
 
 package com.liferay.commerce.internal.security.permission.resource;
 
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
 import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Organization;
 import com.liferay.portal.kernel.security.permission.PermissionChecker;
 import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermissionLogic;
+import com.liferay.portal.kernel.service.OrganizationLocalService;
+
+import java.util.List;
 
 /**
  * @author Riccardo Alberti
  */
 public class CommerceOrderPortletResourcePermissionLogic
 	implements PortletResourcePermissionLogic {
 
+	public CommerceOrderPortletResourcePermissionLogic(
+		OrganizationLocalService organizationLocalService) {
+
+		_organizationLocalService = organizationLocalService;
+	}
+
 	@Override
 	public Boolean contains(
 		PermissionChecker permissionChecker, String name, Group group,
 		String actionId) {
 
-		return permissionChecker.hasPermission(group, name, 0, actionId);
+		if (permissionChecker.hasPermission(group, name, 0, actionId)) {
+			return true;
+		}
+
+		try {
+			List<Organization> organizations =
+				_organizationLocalService.getUserOrganizations(
+					permissionChecker.getUserId(), true);
+
+			for (Organization organization : organizations) {
+				if (permissionChecker.hasPermission(
+						organization.getGroupId(), name, 0, actionId)) {
+
+					return true;
+				}
+			}
+		}
```

Comment:
This logic only checks if the user has the permission (for example the view permission) on a given organization groupId.
But the check if the organization being checked is the one linked to the account "owner" of the order is completely missing. 
This will not work, and will give anyone with the permission on an organization to access all orders, even those about other organizations/accounts.